### PR TITLE
[23.0] vendor: github.com/moby/buildkit v0.10.7-0.20230206124303-b8fdb4b78da0

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to https://github.com/moby/moby/issues/44918
- relates to https://github.com/moby/moby/pull/44920

full diff: https://github.com/moby/buildkit/compare/v0.10.6...b8fdb4b78da0127260ba39b9b7de048cb3877203

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: vendor.mod [hunk 1]
```diff
@@ -49,7 +49,7 @@ require (
 	github.com/klauspost/compress v1.15.12
 	github.com/miekg/dns v1.1.43
 	github.com/mistifyio/go-zfs v2.1.2-0.20190413222219-f784269be439+incompatible
-	github.com/moby/buildkit v0.10.6
+	github.com/moby/buildkit v0.10.7-0.20230206124303-b8fdb4b78da0
```
Comment (thaJeztah) [inline vendor.mod:52]: @tonistiigi do you know if there were other issues for v0.10.x and/or if there were plans for tagging a new version?

**Review by neersighted — APPROVED** [review]

**Comment by thaJeztah** [comment]
Interesting; looks like checkout actions also sets the "safe" option now (or did it already?);

```
Run actions/checkout@v3
Syncing repository: moby/moby
Getting Git version info
  Working directory is 'D:\a\moby\moby\go\src\github.com\docker\docker'
  "C:\Program Files\Git\bin\git.exe" version
  git version 2.39.1.windows.1
Temporarily overriding HOME='D:\a\_temp\216a9447-39cb-41e2-aadd-0d226ca582d6' before making global git config changes
Adding repository directory to the temporary git global config as a safe directory
"C:\Program Files\Git\bin\git.exe" config --global --add safe.directory D:\a\moby\moby\go\src\github.com\docker\docker
```

Looks like we're hitting rate limits now on GitHub though;

```
Fetching the repository
  "C:\Program Files\Git\bin\git.exe" -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +f599f60c9ecf5531eb90cd893b78d09d0b13e8b6:refs/remotes/pull/44936/merge
  Error: fatal: unable to access 'https://github.com/moby/moby/': The requested URL returned error: 429
  The process 'C:\Program Files\Git\bin\git.exe' failed with exit code 128
  Waiting 16 seconds before trying again
  "C:\Program Files\Git\bin\git.exe" -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +f599f60c9ecf5531eb90cd893b78d09d0b13e8b6:refs/remotes/pull/44936/merge
  Error: fatal: unable to access 'https://github.com/moby/moby/': The requested URL returned error: 429
  The process 'C:\Program Files\Git\bin\git.exe' failed with exit code 128
  Waiting 18 seconds before trying again
  "C:\Program Files\Git\bin\git.exe" -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +f599f60c9ecf5531eb90cd893b78d09d0b13e8b6:refs/remotes/pull/44936/merge
  Error: fatal: unable to access 'https://github.com/moby/moby/': The requested URL returned error: 429
  Error: The process 'C:\Program Files\Git\bin\git.exe' failed with exit code 128
```

**Comment by thaJeztah** [comment]
Green now; bringing this one in

**Comment by neersighted** [comment]
The checkout action has set the `safe.directory` option for some time; we only run into this issue when we're in a container, which has a different gitconfig.

**Comment by thaJeztah** [comment]
Yeah, I was wondering if it was something recent (and potentially for the same reason we now also had to set it).

**Pull request merged**